test/puppeteer.spec.js

```diff
@@ -279,6 +279,19 @@ module.exports.addTests = function({testRunner, expect, defaultBrowserOptions})
         expect(page.viewport()).toBe(null);
         await browser.close();
       });
+      it('should take fullPage screenshots when defaultViewport is null', async({server}) => {
+        const options = Object.assign({}, defaultBrowserOptions, {
+          defaultViewport: null
+        });
+        const browser = await puppeteer.launch(options);
+        const page = await browser.newPage();
+        await page.goto(server.PREFIX + '/grid.html');
+        const screenshot = await page.screenshot({
+          fullPage: true
+        });
+        expect(screenshot).toBeGolden('screenshot-grid-fullpage-2.png');
```

Comment:
on a second though, why's screenshot size different here? I thought it shouldn't depend on viewport size at all.

Reply:
Yeah, I was surprised as well ...

Reply:
Original `screenshot-grid-fullpage.png` fixture has 500x1000 dimensions, probably because:
https://github.com/GoogleChrome/puppeteer/blob/5acf953104a09d7d688d76eadced6a4eba75d5d1/test/page.spec.js#L1649

When `viewport` is null, it has default width set to 800 - don't know why

Reply:
Done